api.py

```diff
@@ -130,21 +133,44 @@ def fetch(request: tornado.httpclient.HTTPRequest):
                 self.__class__._last_failed_request = datetime.now()
                 return False
 
-        repo_data, stable_data, \
+        repo_data, stable_data, release_github_data, \
         contributors_data, translations_data = [x.body for x in responses]
 
-        def assemble_release_data(data: str):
-            if isinstance(data, bytes):
-                data = data.decode()
-
-            versions = re.findall("commit=(.*)", data)
+        def assemble_release_data(version_data: str):
+            if isinstance(version_data, bytes):
+                version_data = version_data.decode()
 
+            versions = re.findall("commit=(.*)", version_data)
+            version_codes = re.findall("Build:(.*)", version_data)
+            version_code = version_codes[-1].split(",")[-1]
             return {
                 "version": versions[-1],
+                "version_code": version_code,
+                "apk": "https://f-droid.org/repo/org.schabi.newpipe_" + version_code + ".apk",
             }
 
         repo = json.loads(repo_data)
 
+        # scrap latest GitHub version and apk from website
+        elem = html.fromstring(release_github_data)
```

Comment:
@TheAssassin Do you have any idea how I can reduce the redundancy here? I didn't find a good way to do so.

Reply:
What kind of redundancy are you talking about specifically?

Reply:
We have the same structure for getting the GitHub version, apk and the contributors count: 
Get HTML, use `cssselect` to filter the HTML, check if the we get the a valid number of results, get the first result and modify it.

Unfortunately I couldn't move this to a new method because I failed to execute the steps in the `try` blocks in a good way. My initial idea was to  return `tags[0]`: 
``` python
        def scrape_data_from_html(data, selector: str):
            elem = html.fromstring(data)
            tags = elem.cssselect(selector)
            if len(tags) < 1:
                return -1
            else:
                try:
                    return tags[0]
                except:
                    return -1
```

But than I realised that I'd needed another try and catch to handle the possible exceptions which can be thrown by `tags[0].get('href')`, `tags[0].text` and `int(tags[0].text)`:
```Python
contributors_result = scrape_data_from_html(contributors_data, ".numbers-summary a[href$=contributors] .num")
try:
    contributors_count = int(contributors_result.text)
except:
    contributors_count = -1
```

Reply:
Good,